pydantic/main.py

```diff
@@ -167,7 +167,7 @@ def __new__(mcs, name, bases, namespace):
         class_vars = set()
         # annotation only fields need to come first in fields
         for ann_name, ann_type in annotations.items():
-            if ann_type == ClassVar:
+            if ann_type.__class__ == ClassVar.__class__ or getattr(ann_type, '__origin__', None) == ClassVar:
```

Comment:
I suspect this line breaks the tests. 

``ann_type.__class__`` will return ``typing._SpecialForm`` for ``typing.Any``, ``typing.Union`` and ``typing.ClassVar``, although the latter is the only type I'd like to catch there.

I think we need something like this:

```python
if ann_type == ClassVar or getattr(ann_type, '__origin__', None) == ClassVar:
```